discovery/trends/v0.1/trends.md

```diff
@@ -0,0 +1,355 @@
+fediverse it might be the case that the fediverse server has no content
+at all that references a trending hashtag or link. Or it might not have
+recent content that does so.
+
+To help fediverse servers in this situation discovery FASP include a
+couple of example URIs of content with every result of a query for
+hashtags and links trends. This allows fediverse servers to quickly fill
+their local caches with some content for the received trends. See the
+sections "Requesting Trending Hashtags" and "Requesting Trending Links"
+below for details.
+
+## Requesting Trends
+
+### Common Request Options
+
+All requests for trends from a fediverse server to FASP are HTTP `GET`
+requests that allow - with one exception - using the same optional
+parameters:
+
+* `withinLastHours`: This MUST be a positive integer specifying the
+  number of hours up to the current time for which to compute trends.
+  The minimum value of `1` means records trending within the last hour,
+  while the allowed maximum of `336` means within the last fortnight.
+  If omitted defaults to `24`.
+* `maxCount`: This MUST be a positive integer specifying the maximum
+  number of results that should be returned. If omitted defaults to
+  `20`.
+* `language`: An ISO-639 language code to only receive results in or
+  relevant for the specific language. If omitted results can be in or
+  relevant for any language.
+
+### Common Response Attributes 
+
+All responses include a list of results. Individual result objects
+share the following common keys and values:
+
+* `rank`: A positive integer less than or equal `100` representing the
+  rank of this result. `100` is the highest rank, meaning "most trending".
+  Calculation of the rank is not part of this specification, but within
+  the same FASP software, ranks MUST be comparable. If a fediverse
+  server uses several FASP to query for trends that all run the same
+  software, it MUST be possible to merge results according to rank to
+  get the correct order.
+* `distribution`: An object with one key per hour requested. The keys
+  are consecutive integers each representing one hour. The first hour
+  requested in chronological order is represented by the number `1`. The
+  value of each key is an integer between 0 and 100 representing how
+  much this result was trending in that hour. This information can
+  optionally be used by fediverse servers to display small diagrams
+  ("sparklines") or arrows marking upward and downward trajectories.
+
+### Requesting Trending Content
+
+To request trending content fediverse servers can make an HTTP `GET`
+request to the `/trends/v0/content` endpoint on the FASP.
+
+Example call:
+
+```http
+GET /trends/v0/content
+```
+
+Optionally all the parameters from the previous section "Common Request
+Options" can be used.
+
+Another example, limiting the results to be trending in the last two
+hours and to no more than 10 results:
+
+```http
+GET /trends/v0/content?withinLastHours=2&maxCount=10
```

Comment:
You'd possibly be better using unix epoch here, because servers may have clock skew

Reply:
I am not sure if clock skew should be much of an issue here (unless it is very severe and/or permanent). Worst case the trends are a tad bit older than anticipated.

Some of the reasons for choosing an integer number of hours are:

* The API is explicitly *not* meant to query for historic trend data.
* This effectively sets the resolution of trend data to full hours. This can have a huge impact on the volume of data FASP have to store.